src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -1040,38 +1040,55 @@ public class Program
     public static void Main()
     {
         object o = nameof(Main);
-        Console.WriteLine(o is string { Length is 4 });
-        Console.WriteLine(o is string { NotFound is 4 });
+        Console.WriteLine(o is P { Good is 4 });
+        Console.WriteLine(o is P { NotFound is 4 });
+        Console.WriteLine(o is P { Unreadable is 4 });
+
     }
 }
+class P
+{
+    public int Good = 2;
+    public int Unreadable { set { } }
```

Comment:
Need to test an inaccessible property/field.
Need to test an accessible property with inaccessible getter.